kim/mapper.py

```diff
@@ -90,6 +100,17 @@ def _extract_roles(self, base):
         _roles.update(getattr(cls, 'roles', None) or {})
         _roles.update(getattr(base, '__roles__', None) or {})
 
+        # Roles may be passed as list, convert to whitelist objects in this
+        # case
```

Comment:
Yes I agree, converting to a role is a sensible idea here - It seemed like a nice detail when I wrote it but we should always use role otherwise things will get messy.